### DISCUSSION TITLE
Usage with async-std?

### DISCUSSION THREAD

**Discussion by WhyNotHugo** [hyperium/hyper#3055] [Q&A] [answered]
Can `hyper` somehow be used with `async-std`? A very minimal example indicates I need a tokio runtime:

```rust
use hyper::client::Client;

#[async_std::main]
async fn main() {
    // Still inside `async fn main`...
    let client = Client::new();

    // Parse an `http::Uri`...
    let uri = "http://httpbin.org/ip".parse().unwrap();

    // Await the response...
    let mut resp = client.get(uri).await.unwrap();

    println!("Response: {}", resp.status());
}
```

Yields:

```
thread 'main' panicked at 'there is no reactor running, must be called from the context of a Tokio 1.x runtime', /home/hugo/.local/state/cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.14.23/src/client/connect/dns.rs:121:24
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Given that both runtimes rely on rather standard rust features (`future`, `await/async`, etc), so `hyper` somehow be made to run with tokio? What exactly ties it to one specific runtime?

**Comment by notgull** [comment]
You can, but you have to adapt it manually to the `async-std` runtime. See this example: https://github.com/smol-rs/smol/blob/master/examples/hyper-client.rs

Alternatively, you can use [surf](https://crates.io/crates/surf), which uses `async-std` by default.

**Comment by seanmonstar** [comment, marked as answer]
The default `Client` uses an `HttpConnector` which makes use of `tokio::net::TcpStream`s. You can disable the relevant features and bring your own runtime.

**Reply by WhyNotHugo** [reply]
Right, that makes perfect sense. The `tokio::net::TcpStream` is part of tokio itself.

It's still not fully clear to me why that wouldn't be usable with another runtime though; the underlying futures are the same and all. But I guess that's a separate question.

Thanks for your guidance!
